### PR TITLE
[minor_change] Add resource and datasource for mso_remote_location

### LINKED ISSUE #192: Add `mso_remote_location` resource and data source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add a new resource and data source to manage MSO remote locations.

### PR DESCRIPTION
**Pull request by akinross**
fixes #192

### CODE REVIEW

**Comment by akinross** [comment]
Still requires mso-go-client dependency update to be added before merge, waiting on review and approval.

File: mso/resource_mso_remote_location.go
Comment (sajagana) [inline mso/resource_mso_remote_location.go:110]: This may cause an issue when store_in_statefile is true.

**Review by sajagana — CHANGES_REQUESTED** [review]

File: mso/resource_mso_remote_location.go
Comment (akinross) [reply mso/resource_mso_remote_location.go]: you are correct should be in conditional, changed it

File: website/docs/r/remote_location.html.markdown
Comment (anvitha-jain) [inline website/docs/r/remote_location.html.markdown:79]: Are you sure this is right?

Shouldn't it be like terraform import mso_remote_location.example {remote-location-id}

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by anvitha-jain — DISMISSED** [review]

File: mso/resource_mso_remote_location.go
Comment (shrsr) [inline mso/resource_mso_remote_location.go:68]: Was password intentionally set to computed?

File: mso/resource_mso_remote_location.go
Comment (shrsr) [inline mso/resource_mso_remote_location.go:93]: Suggestion -
"password" or "ssh_key" is required to manage the MSO remote location.

File: website/docs/d/remote_location.html.markdown
Comment (shrsr) [inline website/docs/d/remote_location.html.markdown:18]: name = "remote_location_password" -> "remote_location"

Comment (shrsr) [reply website/docs/d/remote_location.html.markdown:18]: This was a little confusing to read at first

**Review by shrsr — CHANGES_REQUESTED** [review]

**Comment by lhercot** [comment]
Need a rebase

File: mso/resource_mso_remote_location.go
Comment (akinross) [reply mso/resource_mso_remote_location.go:68]: Yes is there a reason it shouldn't?

File: mso/resource_mso_remote_location.go
Comment (akinross) [reply mso/resource_mso_remote_location.go:93]: ok

File: website/docs/d/remote_location.html.markdown
Comment (akinross) [reply website/docs/d/remote_location.html.markdown:18]: what do you mean? I kept different names for the resource on purpose, what name would you propose?

Comment (shrsr) [reply website/docs/d/remote_location.html.markdown:18]:
```
data "mso_remote_location" "password" {
    name = "remote_location_name"
}
```

File: mso/resource_mso_remote_location.go
Comment (shrsr) [reply mso/resource_mso_remote_location.go:68]: It would be good to add **Sensitive: True** here

**Review by shrsr — CHANGES_REQUESTED** [review]

File: mso/resource_mso_remote_location.go
Comment (akinross) [reply mso/resource_mso_remote_location.go:68]: ok added for password, sshkey and passphrase

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**
